src/containers/pages/MDALite/cddReports/supervisors.tsx

```diff
@@ -0,0 +1,199 @@
+  reducerName as wardsReducerName,
+} from '../../../../store/ducks/superset/MDALite/wards';
+import { getCddTableProps, supervisorColumns } from './helpers';
+
+/** register the reducers */
+reducerRegistry.register(supervisorReducerName, supervisorReducer);
+reducerRegistry.register(wardsReducerName, wardsReducer);
+
+/** declear selectors */
+const supervisorsArraySelector = makeMDALiteSupervisorsArraySelector();
+const makeMDALiteWardsSelector = makeMDALiteWardsArraySelector();
+
+/** MDA-Lite supervisor reports props */
+interface MDALiteSupervisorReportsProps {
+  fetchSupervisors: typeof fetchMDALiteSupervisors;
+  fetchWards: typeof fetchMDALiteWards;
+  service: typeof supersetFetch;
+  supervisorData: MDALiteSupervisor[];
+  wardData: MDALiteWards[];
+}
+
+/** MDA-Lite supervisor reports table component */
+const MDALiteSupervisorReports = (
+  props: MDALiteSupervisorReportsProps & RouteComponentProps<RouteParams>
+) => {
+  const { supervisorData, wardData, service, fetchWards, fetchSupervisors } = props;
+
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const { params } = props.match;
+  const { planId, jurisdictionId, supervisorId } = params;
+  const wardName = wardData[0]?.ward_name;
+
+  async function loadData() {
+    setLoading(supervisorData.length < 1);
+    try {
+      const planIdFilter: SupersetAdhocFilterOption[] = [];
+      const jurisdictionFilter: SupersetAdhocFilterOption[] = [];
+      if (jurisdictionId) {
+        jurisdictionFilter.push({
+          comparator: jurisdictionId,
+          operator: '==',
+          subject: 'base_entity_id',
+        });
+      }
+      if (planId) {
+        planIdFilter.push({ comparator: planId, operator: '==', subject: 'plan_id' });
+      }
+      // get supervisors data
+      const fetchDataParams = superset.getFormData(SUPERSET_MAX_RECORDS, [
+        ...planIdFilter,
+        ...jurisdictionFilter,
+      ]);
+      await service(SUPERSET_MDA_LITE_REPORTING_SUPERVISORS_DATA_SLICE, fetchDataParams).then(res =>
+        fetchSupervisors(res)
+      );
+      // get ward data
+      if (!wardData.length) {
+        await service(SUPERSET_MDA_LITE_REPORTING_WARD_SLICE, fetchDataParams).then(res =>
+          fetchWards(res)
+        );
+      }
+    } catch (e) {
+      displayError(e);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  useEffect(() => {
+    loadData().catch(err => displayError(err));
+  }, [planId, jurisdictionId, supervisorId]);
+
+  const currentPage = {
+    label: wardName,
+    url: '',
+  };
+  const breadcrumbProps = {
+    currentPage,
+    pages: [
+      {
+        label: '...',
+        url: `${REPORT_MDA_LITE_WARD_URL}/${planId}/${wardData[0]?.parent_id}`,
+      },
+    ],
+  };
+  const currentTitle = wardName
+    ? `${MDA_LITE_REPORTING_TITLE}: ${wardName}`
+    : MDA_LITE_REPORTING_TITLE;
+
+  // table props
+  const tableProps = getCddTableProps(supervisorColumns, supervisorData, props);
+
+  if (loading) {
```

Comment:
Shouldn't we have this piece of code above the code that precedes it. i.e Why should we execute code then check for the loading status? In a case where loading is false, all that execution would be futile